operational/azure/total_instance_usage_report/azure_total_instance_usage_report.pt

```diff
@@ -0,0 +1,498 @@
+    { "dimension": "category", "type": "equal", "value": "Compute" },
+    { "dimension": "resource_type", "type": "substring", "substring": "Virtual Machines" },
+    { "dimension": "vendor", "type": "substring", substring: "Azure" }, // Use a substring type to include both "Azure" (Azure EA) and "AzureMCA-Enterprise" (Azure MCA) subscriptions
+    {
+      "type": "not",
+      "expression": { "dimension": "instance_type", "type": "equal", "value": "None" }
+    }
+  ])
+
+  // Payload to send to API
+  payload = {
+    dimensions: [ "billing_center_id", "instance_type", "usage_unit" ],
+    billing_center_ids: ds_billing_centers_filtered,
+    filter: { type: "and", expressions: expressions },
+    metrics: [ "cost_nonamortized_unblended_adj", "usage_amount" ],
+    granularity: "month",
+    end_at: end_date,
+    start_at: start_date
+  }
+
+  // Request
+  var request = {
+    auth: "auth_flexera",
+    verb: "POST",
+    host: rs_optima_host,
+    path: "/bill-analysis/orgs/" + rs_org_id + "/costs/aggregated",
+    headers: {
+      "User-Agent": "RS Policies",
+    },
+    body_fields: payload
+  }
+  EOS
+end
+
+datasource "ds_azure_instance_size_map" do
+  request do
+    verb "GET"
+    host "raw.githubusercontent.com"
+    path "/flexera-public/policy_templates/master/data/azure/instance_types.json"
+    header "User-Agent", "RS Policies"
+  end
+end
+
+datasource "ds_isf_ratio_csv" do
+  request do
+    host "aka.ms"
+    path "/isf"
+  end
+  result do
+    encoding "text"
+  end
+end
+
+datasource "ds_instance_units_per_family" do
+  run_script $js_instance_units_per_family, $ds_azure_instance_size_map, $ds_isf_ratio_csv, $ds_usage_data, $param_unit_time, $param_unit_instance
+end
+
+script "js_instance_units_per_family", type: "javascript" do
+  parameters "ds_azure_instance_size_map", "ds_isf_ratio_csv", "ds_usage_data", "param_unit_time", "param_unit_instance"
+  result "result"
+  code <<-'EOS'
+
+  // Convert Instance Family csv to json
+  var isf_ratio_array = ds_isf_ratio_csv.toString().split("\r\n")
+  var inst_families = []
+  _.each(isf_ratio_array, function(ratio) {
+    inst_families.push({
+      "instance_family": ratio.split(",")[0],
+      "instance_type": ratio.split(",")[1]
+    })
+  })
+
+  // Tables to convert parameters to units
+  time_unit = { "Hours": 1, "Days": 24, "Weeks": 168, "Months": 730, "Years": 8760 }
+  instance_unit = { "Normalized Units (NFUs)": "nfu", "vCPUs": "vcpu", "Memory (GiB)": "memory" }
+  result = []
+
+  // Get Normalization Factor units from Instance Data json
+  cpu_ratios = {}
+
+  _.each(_.keys(ds_azure_instance_size_map), function(instance_type) {
+    cpu = ds_azure_instance_size_map[instance_type][instance_unit[param_unit_instance]]
+    if (typeof(cpu) != 'number' && typeof(cpu) != 'string') { cpu = 1 }
+    cpu_ratios[instance_type] = Number(cpu)
+  })
+
+  // Enrich current data with Instance Family, and then calculate Normalized Instance Hours using NFU
```

Comment:
Similar to other places, we shouldn't use "NFU", which is AWS specific, when discussing Azure